File: FinalLab/native/VolunteersApp/app/src/main/java/com/example/ma/volunteersapp/CustomCursorAdapter.java
package com.example.ma.volunteersapp;

import android.annotation.TargetApi;
import android.content.Context;
import android.database.Cursor;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.CursorAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class CustomCursorAdapter extends CursorAdapter {

    private LayoutInflater layoutInflater;

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    public CustomCursorAdapter(Context context, Cursor c, List items, int flags) {
        super(context, c, flags);
        layoutInflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
        View v = layoutInflater.inflate(R.layout.row_volunteers, viewGroup, false);
        MyHolder holder = new MyHolder();
        holder.ListID = (TextView)v.findViewById(R.id.listID);
        holder.ListName = (TextView)v.findViewById(R.id.listName);
        holder.ListPhone = (TextView)v.findViewById(R.id.listPhone);
        holder.ListCreated = (TextView)v.findViewById(R.id.listCreated);
        v.setTag(holder);
        return v;
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        MyHolder holder = (MyHolder)view.getTag();

        holder.ListID.setText(cursor.getString(cursor.getColumnIndex(DBHelper.row_id)));
        holder.ListName.setText(cursor.getString(cursor.getColumnIndex(DBHelper.row_name)));
        holder.ListPhone.setText(cursor.getString(cursor.getColumnIndex(DBHelper.row_phone)));
        holder.ListCreated.setText(cursor.getString(cursor.getColumnIndex(DBHelper.row_created)));
    }

    class MyHolder{
        TextView ListID;
        TextView ListName;
        TextView ListPhone;
        TextView ListCreated;
    }
}

File: Labs/Lab1/ManageVolunteersNativeApplicationKotlin/app/src/main/java/com/example/myapplication/CustomCursorAdapter.kt
package com.example.myapplication

import android.annotation.TargetApi
import android.content.Context
import android.database.Cursor
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import android.widget.TextView

class CustomCursorAdapter @TargetApi(Build.VERSION_CODES.HONEYCOMB) constructor(
    context: Context,
    c: Cursor?,
    flags: Int
) : CursorAdapter(context, c, flags) {
    private val layoutInflater: LayoutInflater
    override fun newView(
        context: Context,
        cursor: Cursor,
        viewGroup: ViewGroup
    ): View {
        val v: View = layoutInflater.inflate(R.layout.row_notes, viewGroup, false)
        val holder = MyHolder()
        holder.ListID = v.findViewById<View>(R.id.listID) as TextView
        holder.ListTitle = v.findViewById<View>(R.id.listTitle) as TextView
        holder.ListDetail = v.findViewById<View>(R.id.listDetail) as TextView
        holder.ListCreated = v.findViewById<View>(R.id.listCreated) as TextView
        v.tag = holder
        return v
    }

    override fun bindView(
        view: View,
        context: Context,
        cursor: Cursor
    ) {
        val holder = view.tag as MyHolder
        holder.ListID!!.text = cursor.getString(cursor.getColumnIndex(DBHelper.row_id))
        holder.ListTitle!!.text = cursor.getString(cursor.getColumnIndex(DBHelper.row_title))
        holder.ListDetail!!.text = cursor.getString(cursor.getColumnIndex(DBHelper.row_note))
        holder.ListCreated!!.text = cursor.getString(cursor.getColumnIndex(DBHelper.row_created))
    }

    internal inner class MyHolder {
        var ListID: TextView? = null
        var ListTitle: TextView? = null
        var ListDetail: TextView? = null
        var ListCreated: TextView? = null
    }

    init {
        layoutInflater =
            context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }
}